src/coord/rmr/rmr.c

```diff
@@ -442,7 +447,9 @@ struct MRIteratorCtx {
   short pending;    // Number of shards with more results (not depleted)
   short inProcess;  // Number of currently running commands on shards
   bool timedOut;    // whether the coordinator experienced a timeout
-  bool freeFlag;    // whether the MRIterator should be freed
+  // reference counter of the iterator.
+  // When it reaches 0, both readers and the writer agree that the iterator can be released
+  int8_t itRefCount;
```

Comment:
Using int8 here implies that we assume that there will be no more than 128 shards. Not sure it is a valid assumption

Reply:
There shouldn't be more than 3